app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -468,14 +468,15 @@ const Topic = RestModel.extend({
             this.siteSettings.can_permanently_delete && deleted_by.admin,
         });
         if (
-          !deleted_by.staff &&
-          !deleted_by.groups.some(
-            (group) => group.name === this.category.reviewable_by_group_name
-          ) &&
-          !(
-            this.siteSettings.tl4_delete_posts_and_topics &&
-            deleted_by.trust_level >= 4
-          )
+          opts.force_destroy ||
```

Comment:
That's a very complicated set of conditions. We should probably extract those in variables so it's easier to read/understand.